### DISCUSSION TITLE
alternative for mmap

### DISCUSSION THREAD

**Discussion by ghost** [axboe/liburing#858] [Q&A]
Hi,

I am currently seeking a viable alternative to replace mmap in an application after reading
[mmap analysis in DB apps](https://db.cs.cmu.edu/mmap-cidr2022/) 
Mmap shares a database-ish file among multiple threads, Read/write, and some atomic conditional variables enable reads/writes to some parts of the file. Those condvars are stored in the shared memory.

After writing, I use a msync.

It works perfectly fine as it is with mmap. But I am eager to learn new things.

BTW, I use a fast cache before entering the DB. The issue being that it is redundant with mmap/ the OS, since we can't disable mmap cahing. That is a useless performance degradation.

After deciding that direct/IO is not a solution for me because of the -very stringent- alignment constraints, may io_uring be of any help?

I don't think so, I read the tutorial examples, but maybe someone who knows enough could give me some pointers.

Thanks

**Comment by Dan-Do** [comment]
Hello @ghost,
How is thing going on your finding? Could you share your thoughts?
Thanks
